tests/openwisp2/integrations/tests.py

```diff
@@ -60,7 +61,7 @@ def test_captive_portal_login_mock(self, *args, **kwargs):
             self.assertIsNone(ra.stop_time)
             self.assertEqual(ra.username, user.username)
 
-        return radius_token, ra
+        cache.delete(f'rt-{radius_token.user.username}')
```

Comment:
So, the tests are failing due to cache of token. I had spent many hours in trying to resolve this :sweat_smile:.
Thanks for fixing.

Reply:
Yes, it took me couple of hours to understand what was happening.